src/toxcore/dht/client.rs

```diff
@@ -53,6 +54,8 @@ pub struct Client {
     pub precomputed_key: PrecomputedKey,
     /// shaed mpsc tx part
     pub tx: Tx,
+    /// last received ping-response time
+    pub last_resp_time: Instant
```

Comment:
@kurnevsky should we use SystemTime or Instant here?

Reply:
Another solution - if we pass timeout as argument (as you proposed) we can set it to 0 in tests to expire all clients.